File: Human_written_code/Code_with_comments/CPP/Q22.cpp
#include <iostream>
using namespace std;

void solve() {
    long long a, b, c, d;
    cin >> a >> b >> c >> d;  // Read four integers
    long long x = a * d, y = b * c;  // Compute cross-products

    if (x == y)  // Check if products are equal
        cout << "0\n";  // Output 0 if ratios are equal
    else if ((y != 0 && x % y == 0) || (x != 0 && y % x == 0))
        cout << "1\n";  // Output 1 if one ratio is a multiple of the other
    else
        cout << "2\n";  // Output 2 if ratios are neither equal nor multiples
}

int main() {
    ios_base::sync_with_stdio(false);  // Optimize input/output speed
    cin.tie(nullptr);  // Untie cin and cout for faster IO

    int t;
    cin >> t;  // Read number of test cases
    while (t--)
        solve();  // Solve each test case
}

File: Human_written_code/Code_with_comments/Python/Q22.py
def solve():
    # Read four integers a, b, c, d from input
    a, b, c, d = map(int, input().split())
    
    # Calculate x as the product of a and d
    x = a * d
    # Calculate y as the product of b and c
    y = b * c

    # Check if x and y are equal, meaning the fractions are equivalent
    if x == y:
        print(0)  # Output 0 if the two fractions are already equal
    # Check if one value is a multiple of the other
    elif (y != 0 and x % y == 0) or (x != 0 and y % x == 0):
        print(1)  # Output 1 if one multiplication will make them equal
    else:
        print(2)  # Output 2 if at least two steps are needed to make them equal

def main():
    # Read the number of test cases
    T = int(input())
    # Execute the solve function for each test case
    for _ in range(T):
        solve()

# Entry point of the script
if __name__ == "__main__":
    main()
